app/client/components/delivery/records/deliveryRecords.tsx

```diff
@@ -1,77 +1,591 @@
+      trackEvent({
+        eventCategory: "delivery-problem",
+        eventAction: "continue_to_step_2_button_click",
+        product: {
+          productType: props.routeableStepProps.productType,
+          productDetail: props.productDetail
+        },
+        eventLabel: props.routeableStepProps.productType.urlPart
+      });
+      setPageStatus(PageStatus.REPORT_ISSUE_STEP_2);
+    }
+  };
+
+  const addRecordToDeliveryProblem = (id: string) =>
+    setSelectedProblemRecords([...selectedProblemRecords, id]);
+  const removeRecordFromDeliveryProblem = (id: string) =>
+    setSelectedProblemRecords(
+      selectedProblemRecords.filter((existingId: string) => existingId !== id)
+    );
+  const resultsPerPage = 7;
+  const totalPages = Math.ceil(props.data.results.length / resultsPerPage);
+  const scrollToTop = () => window.scrollTo(0, 0);
+  const resetDeliveryRecordsPage = () => setPageStatus(PageStatus.READ_ONLY);
+
+  const filterData = (productPartName: ProductUrlPart) => {
+    const NumOfRecordsToShow =
+      ProductTypes[productPartName as ProductTypeKeys].delivery?.records
+        ?.numberOfProblemRecordsToShow || props.data.results.length;
+
+    if (pageStatus !== PageStatus.READ_ONLY) {
+      return props.data.results
+        .filter(record => !record.problemCaseId)
+        .slice(0, NumOfRecordsToShow - 1);
+    }
+    return props.data.results.filter((element, index) =>
+      isRecordInCurrentPage(
+        index,
+        currentPage * resultsPerPage,
+        currentPage * resultsPerPage + resultsPerPage - 1
+      )
+    );
+  };
+
+  const capatalize = (input: string) =>
+    input.charAt(0).toUpperCase() + input.slice(1);
+
+  const isRecordInCurrentPage = (
+    index: number,
+    currentPageStartIndex: number,
+    currentPageEndIndex: number
+  ) => index >= currentPageStartIndex && index <= currentPageEndIndex;
+
+  const hasExistingDeliveryProblem =
+    props.data.results.findIndex(
+      deliveryRecord => deliveryRecord.problemCaseId
+    ) > -1;
+
+  const filteredData = filterData(props.routeableStepProps.productType.urlPart);
+
+  return (
+    <DeliveryRecordsProblemContext.Provider
+      value={{
+        subscription: props.productDetail.subscription,
+        subscriptionCurrency: props.subscriptionCurrency,
+        productName: capatalize(
+          props.routeableStepProps.productType.shortenedFriendlyName ||
+            props.routeableStepProps.productType.friendlyName
+        ),
+        apiProductName:
+          props.routeableStepProps.productType.fulfilmentDateCalculator
+            ?.productFilenamePart,
+        problemType: deliveryProblem,
+        affectedRecords: props.data.results.filter(record =>
+          selectedProblemRecords.includes(record.id)
+        ),
+        deliveryProblemMap: props.data.deliveryProblemMap,
+        isTestUser: props.productDetail.isTestUser,
+        showProblemCredit:
+          !props.productDetail.subscription.cancelledAt &&
```

Comment:
We have a requirement that missed along the way...
```suggestion
          !props.productDetail.subscription.cancelledAt &&
          props.productDetail.subscription.autoRenew &&
```